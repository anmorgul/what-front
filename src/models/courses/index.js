export { coursesReducer } from './reducer.js';
export { coursesWatcher, fetchActiveCourses, fetchNotActiveCourses, createCourse, editCourse, deleteCourse, reactivateCourse } from './actions.js';
export { coursesActiveSelector, coursesNotActiveSelector, createdCourseSelector, editedCourseSelector, deletedCourseSelector, reactivatedCourseSelector } from './selectors.js';