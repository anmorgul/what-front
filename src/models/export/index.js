export { exportReducer } from './reducer.js';
export { exportStudentsResults, exportStudentsClassbook, exportStudentClassbook, exportStudentGroupResults, exportStudentResults, exportWatcher } from './actions.js';
export { exportStudentClassbookSelector, exportStudentGroupResultsSelector, exportStudentResultSelector, exportStudentsClassbookSelector, exportStudentsResultSelector } from './selectors.js';