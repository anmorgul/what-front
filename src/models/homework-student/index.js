export { homeworkStudentReducer } from './reducer.js';
export { getHomeworkStudent, getHomeworkByIdStudent, addHomeworkStudent, updateHomeworkStudent } from './actions.js';
export {
  homeworkStudentSelector,
  homeworkByIdStudentSelector,
  addedHomeworkStudentSelector,
  updatedHomeworkStudentSelector,
} from './selectors.js';
