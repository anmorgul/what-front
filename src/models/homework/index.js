export { homeworkReducer } from './reducer.js';
export { getHomework, addHomework, updateHomework } from './actions.js';
export {
  homeworkSelector,
  addedHomeworkSelector,
  updatedHomeworkSelector,
} from './selectors.js';
