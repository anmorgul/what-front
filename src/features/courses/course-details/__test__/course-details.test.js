import React from 'react';
import { useSelector } from 'react-redux';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { BrowserRouter, Router, useHistory } from 'react-router-dom';

import { currentUserSelector } from '@/models';
import { homepages, paths } from '@/shared';
import { CourseDetails } from '../course-details';

jest.mock('react-redux', () => ({
  useSelector: jest.fn().mockReturnValue({
    currentUser: jest.fn().mockReturnValue({
      role: 4,
    }),
  }),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn().mockReturnValue({ push: jest.fn() }),
}));

describe('Render of CourseDetails', () => {
  let mockCoursesDetailsSelector;
  let ID;

  beforeEach(() => {
    mockCoursesDetailsSelector = {
      data: [],
      isLoading: false,
      loaded: true,
      error: '',
      currentUser: { role: 4 },
    };

    ID = 1;

    useSelector.mockReturnValueOnce(mockCoursesDetailsSelector);
  });

  it('Should the component be rendered.', () => {
    const { container } = render(
      <BrowserRouter>
        <CourseDetails id={ID} coursesData={mockCoursesDetailsSelector} />
      </BrowserRouter>
    );

    const courseDetailsContainer =
      container.getElementsByClassName('container');

    expect(courseDetailsContainer).toMatchSnapshot();
  });

  it('Should the user navigates to page by role.', () => {
    const history = useHistory();
    const { currentUser } = useSelector(currentUserSelector);

    render(
      <BrowserRouter>
        <CourseDetails id={ID} coursesData={mockCoursesDetailsSelector} />
      </BrowserRouter>
    );

    history.push(homepages[currentUser.role]);

    expect(history.push).toHaveBeenCalledWith(homepages[currentUser.role]);
  });

  it('Should the current user does not exist.', async () => {
    const history = createMemoryHistory();

    useSelector.mockReturnValue({
      currentUser: null,
    });

    render(
      <Router history={history}>
        <CourseDetails id={ID} coursesData={mockCoursesDetailsSelector} />
      </Router>
    );

    history.push(paths.NOT_FOUND);

    expect(history.location.pathname).toBe(paths.NOT_FOUND);
  });

  it('Should redirect to 404 page in case of error in course details data.', () => {
    const history = createMemoryHistory();

    useSelector.mockReturnValue({
      ...mockCoursesDetailsSelector,
      error: 'Something went wrong',
      loaded: false,
    });

    render(
      <Router history={history}>
        <CourseDetails id={ID} coursesData={mockCoursesDetailsSelector} />
      </Router>
    );

    history.push(paths.NOT_FOUND);

    expect(history.location.pathname).toBe(paths.NOT_FOUND);
  });
});
