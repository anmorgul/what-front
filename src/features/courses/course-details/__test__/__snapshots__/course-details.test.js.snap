// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render of CourseDetails Should the component be rendered. 1`] = `
HTMLCollection [
  <div
    class="container"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-sm-12 card shadow col-md-12"
      >
        <div
          class="px-2 py-4"
        >
          <h3>
            Course Details
          </h3>
          <hr />
          <div
            class="row"
          >
            <div
              class="col-12 col-md-6 font-weight-bolder"
            >
              <span>
                Course name: 
              </span>
            </div>
            <div
              class="col-12 col-md-6"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
